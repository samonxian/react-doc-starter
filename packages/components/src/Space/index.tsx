import React from 'react';
import { Space, SpaceProps } from 'antd';

/**
 * 基于 antd 适配为 sec 的格式
 */
const RSpace: {
  (props: SpaceProps): JSX.Element;
  defaultProps?: Record<string, any>;
  displayName: string;
} = (props: SpaceProps) => {
  const { children, direction } = props;
  let { style = {} } = props;

  if (direction === 'vertical') {
    style = { ...style, width: '100%', display: 'flex' };
  }

  return (
    <Space {...props} direction={direction} style={style}>
      {children}
    </Space>
  );
};

RSpace.displayName = 'RSpace';

export default RSpace;
