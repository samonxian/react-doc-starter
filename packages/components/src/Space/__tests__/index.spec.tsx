import { expect, it } from 'vitest';
import React from 'react';
import renderer from 'react-test-renderer';
import Space from '../index';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

it('Space rendered', () => {
  const component = renderer.create(
    <Space>
      <div>1</div>
      <div>2</div>
    </Space>,
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
