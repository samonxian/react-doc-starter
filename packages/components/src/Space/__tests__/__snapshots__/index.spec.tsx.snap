// Vitest Snapshot v1

exports[`Space rendered 1`] = `
<div
  className="ant-space ant-space-horizontal ant-space-align-center"
  style={{}}
>
  <div
    className="ant-space-item"
    style={
      {
        "marginRight": 8,
      }
    }
  >
    <div>
      1
    </div>
  </div>
  <div
    className="ant-space-item"
    style={{}}
  >
    <div>
      2
    </div>
  </div>
</div>
`;
